### ISSUE TITLE
Dependency Update:  typing_extensions>=3.7.4.3

### ISSUE THREAD

**Issue by Chris-Olsen** [SeldonIO/alibi#600]
Anything less breaks huggingface_hub.

**Comment by mauicv** [comment]
Hey @Chris-Olsen,
Thanks for opening the issue. 

I think this is related to #587. Can you share with me the version of python your using and what the error is that you get? `huggingface_hub` is a dependency of `transformers` which we depend on, but it specifies [`"typing-extensions>=3.7.4.3"`](https://github.com/huggingface/huggingface_hub/blob/d5c4d5934ff9ee7186ccfb024d313b04ed93c36c/setup.py#L19) so I'm trying to figure out how users are getting versions of `typing_extensions` less than `3.7.4.3`.

**Comment by Chris-Olsen** [comment]
Hey mauicv,

I am using python 3.8.6 at the moment.  For some reason huggingface_hub was unable to handle typing_extensions less than 3.7.4.3.  I found that I had 3.7.4.2 and that updating typing_extensions to the latest version resolved the error.  I only want to point out that the dependencies might need to be updated for alibi so that the examples can work.  I experienced the previous error simply upon 'import alibi'.

I hope this helps and I'd be glad to help test things for you.


Best Regards,

Christopher Olsen, PhD
Staff Data Scientist
Data Analytics, Applied Digital Technologies
ConocoPhillips
925 N. Eldridge Parkway
Houston, TX 77079
281-293-4323

From: mauicv ***@***.***>
Sent: Monday, February 28, 2022 7:43 AM
To: SeldonIO/alibi ***@***.***>
Cc: Olsen, Chris S ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL]Re: [SeldonIO/alibi] Dependency Update: typing_extensions>=3.7.4.3 (Issue #600)

CAUTION:This email originated from outside of the organization. Do not click links or open attachments unless you recognize the sender and know the content is safe.




Hey @Chris-Olsen<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FChris-Olsen&data=04%7C01%7Cchristopher.s.olsen%40conocophillips.com%7C68263af0c82b490308a608d9fac037ea%7Cb449db5ea80a48eba4c23c88bb78353b%7C0%7C0%7C637816525805796406%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=DP8B87jOwnyXbaHFp8JMnzrFB5T%2BgOzg4%2FywPzFvLow%3D&reserved=0>,
Thanks for opening the issue.

I think this is related to #587<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FSeldonIO%2Falibi%2Fissues%2F587&data=04%7C01%7Cchristopher.s.olsen%40conocophillips.com%7C68263af0c82b490308a608d9fac037ea%7Cb449db5ea80a48eba4c23c88bb78353b%7C0%7C0%7C637816525805796406%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=Zsvo6t1MfkINNXS1dlqPsA6W53O3xvM8WCDdA3oxHA4%3D&reserved=0>. Can you share with me the version of python your using and what the error is that you get? huggingface_hub is a dependency of transformers which we depend on, but it specifies "typing-extensions>=3.7.4.3"<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fhuggingface%2Fhuggingface_hub%2Fblob%2Fd5c4d5934ff9ee7186ccfb024d313b04ed93c36c%2Fsetup.py%23L19&data=04%7C01%7Cchristopher.s.olsen%40conocophillips.com%7C68263af0c82b490308a608d9fac037ea%7Cb449db5ea80a48eba4c23c88bb78353b%7C0%7C0%7C637816525805796406%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=h987uw%2BmnPD21nxRjwKNTDzIhwymVN0m3MpYH8lIR0A%3D&reserved=0> so I'm trying to figure out how users are getting versions of typing_extensions less than 3.7.4.3.

-
Reply to this email directly, view it on GitHub<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FSeldonIO%2Falibi%2Fissues%2F600%23issuecomment-1054269613&data=04%7C01%7Cchristopher.s.olsen%40conocophillips.com%7C68263af0c82b490308a608d9fac037ea%7Cb449db5ea80a48eba4c23c88bb78353b%7C0%7C0%7C637816525805796406%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=VO7gMj77TauOxaoI%2Fm8Cm4z2H%2BYXIgJtyZ3miaQtZz4%3D&reserved=0>, or unsubscribe<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAPDQWEBLXF25OSSSNEGCGJLU5N3VZANCNFSM5PIJDNZA&data=04%7C01%7Cchristopher.s.olsen%40conocophillips.com%7C68263af0c82b490308a608d9fac037ea%7Cb449db5ea80a48eba4c23c88bb78353b%7C0%7C0%7C637816525805796406%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=SJsBlNyDsBsNP0kl6EluYv86mpzDuMwweaeCod9iGYk%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7Cchristopher.s.olsen%40conocophillips.com%7C68263af0c82b490308a608d9fac037ea%7Cb449db5ea80a48eba4c23c88bb78353b%7C0%7C0%7C637816525805796406%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=FnntxRJf1LSJYYV%2FIdaTZFPoQpKksBOcgIzxVfZ8fIM%3D&reserved=0> or Android<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26referrer%3Dutm_campaign%253Dnotification-email%2526utm_medium%253Demail%2526utm_source%253Dgithub&data=04%7C01%7Cchristopher.s.olsen%40conocophillips.com%7C68263af0c82b490308a608d9fac037ea%7Cb449db5ea80a48eba4c23c88bb78353b%7C0%7C0%7C637816525805796406%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=LtZqB%2FKlq4514STEIuh8qsYcTIPPYMO7TInuTZ%2BvBAY%3D&reserved=0>.

**Comment by mauicv** [comment]
Hey @Chris-Olsen 

> Hey mauicv,
I am using python 3.8.6 at the moment.  For some reason huggingface_hub was unable to handle typing_extensions less than 3.7.4.3.  I found that I had 3.7.4.2 and that updating typing_extensions to the latest version resolved the error.  I only want to point out that the dependencies might need to be updated for alibi so that the examples can work.  I experienced the previous error simply upon 'import alibi'.
I hope this helps and I'd be glad to help test things for you.

Thanks for this @Chris-Olsen. Yeah `huggingface-hub` requires `TypeAlias` from `typing-extensions` which is only added in 3.7.4.3. This dependency version should be enforced by `huggingface-hub` though. Would it be possible for you to send me your output of `pip list` in your virtual environment?

Basically, we'd rather not update our version bounds to account for other libraries' dependency requirements because it's really their responsibility and it becomes unmaintainable for us.

**Comment by Chris-Olsen** [comment]
Sure.  Not sure why it wasn't caught by huggingface-hub.


Best Regards,

Christopher Olsen, PhD
Staff Data Scientist
Data Analytics, Applied Digital Technologies
ConocoPhillips
925 N. Eldridge Parkway
Houston, TX 77079
281-293-4323

From: mauicv ***@***.***>
Sent: Wednesday, March 2, 2022 5:03 AM
To: SeldonIO/alibi ***@***.***>
Cc: Olsen, Chris S ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL]Re: [SeldonIO/alibi] Dependency Update: typing_extensions>=3.7.4.3 (Issue #600)

CAUTION:This email originated from outside of the organization. Do not click links or open attachments unless you recognize the sender and know the content is safe.




Hey @Chris-Olsen<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FChris-Olsen&data=04%7C01%7Cchristopher.s.olsen%40conocophillips.com%7C29c86dfe2c1d458c7bcb08d9fc3c284b%7Cb449db5ea80a48eba4c23c88bb78353b%7C0%7C0%7C637818157578497100%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=e7xLM8d81q%2FtdXIPw4toPRLvC7R4fUA5322KtL17RwI%3D&reserved=0>

Hey mauicv,
I am using python 3.8.6 at the moment. For some reason huggingface_hub was unable to handle typing_extensions less than 3.7.4.3. I found that I had 3.7.4.2 and that updating typing_extensions to the latest version resolved the error. I only want to point out that the dependencies might need to be updated for alibi so that the examples can work. I experienced the previous error simply upon 'import alibi'.
I hope this helps and I'd be glad to help test things for you.

Thanks for this @Chris-Olsen<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FChris-Olsen&data=04%7C01%7Cchristopher.s.olsen%40conocophillips.com%7C29c86dfe2c1d458c7bcb08d9fc3c284b%7Cb449db5ea80a48eba4c23c88bb78353b%7C0%7C0%7C637818157578497100%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=e7xLM8d81q%2FtdXIPw4toPRLvC7R4fUA5322KtL17RwI%3D&reserved=0>. Yeah huggingface-hub requires TypeAlias from typing-extensions which is only added in 3.7.4.3. This dependency version should be enforced by huggingface-hub though. Would it be possible for you to send me your output of pip list in your virtual environment?

Basically, we'd rather not update our version bounds to account for other libraries' dependency requirements because it's really their responsibility and it becomes unmaintainable for us.

-
Reply to this email directly, view it on GitHub<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FSeldonIO%2Falibi%2Fissues%2F600%23issuecomment-1056795514&data=04%7C01%7Cchristopher.s.olsen%40conocophillips.com%7C29c86dfe2c1d458c7bcb08d9fc3c284b%7Cb449db5ea80a48eba4c23c88bb78353b%7C0%7C0%7C637818157578497100%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=NPi7ZY%2F%2B68DUsMdJ455gaaaMptSTbaBW2y2GzdiLdpI%3D&reserved=0>, or unsubscribe<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAPDQWEBGJN3ERP4OXB7EF53U55DEVANCNFSM5PIJDNZA&data=04%7C01%7Cchristopher.s.olsen%40conocophillips.com%7C29c86dfe2c1d458c7bcb08d9fc3c284b%7Cb449db5ea80a48eba4c23c88bb78353b%7C0%7C0%7C637818157578653319%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=ZwKu6MUgpXifjLg%2FUHdsppsst2zQVzpqHIaUPWje5oU%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7Cchristopher.s.olsen%40conocophillips.com%7C29c86dfe2c1d458c7bcb08d9fc3c284b%7Cb449db5ea80a48eba4c23c88bb78353b%7C0%7C0%7C637818157578653319%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=XtnVrk4aSA8%2FwfDYtZloQ526MuiKwAPZZZgosOmIOlA%3D&reserved=0> or Android<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26referrer%3Dutm_campaign%253Dnotification-email%2526utm_medium%253Demail%2526utm_source%253Dgithub&data=04%7C01%7Cchristopher.s.olsen%40conocophillips.com%7C29c86dfe2c1d458c7bcb08d9fc3c284b%7Cb449db5ea80a48eba4c23c88bb78353b%7C0%7C0%7C637818157578653319%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=Io7cZ0%2BEWEP4uBrmxZjTcxplrOovsRnw9LSaNYKfBuI%3D&reserved=0>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

Package                            Version
---------------------------------- -------------------
-ensorflow                         2.2.0
-stropy                            4.0.1.post1
-umpy                              1.18.5
absl-py                            0.10.0
access                             1.1.3
affine                             2.3.0
aiohttp                            3.6.3
alabaster                          0.7.12
alembic                            1.4.1
alibi                              0.6.4
altair                             4.1.0
amply                              0.1.4
anaconda-client                    1.7.2
anaconda-navigator                 1.9.12
anaconda-project                   0.8.3
ansiwrap                           0.8.4
appdirs                            1.4.4
argh                               0.26.2
arn                                0.1.5
arrow                              1.0.3
asn1crypto                         1.3.0
astor                              0.8.1
astroid                            2.4.2
astropy                            5.0
astunparse                         1.6.3
async-generator                    1.10
async-timeout                      3.0.1
atomicwrites                       1.4.0
attrs                              19.3.0
autopep8                           1.5.3
azure-core                         1.8.2
azure-storage-blob                 12.5.0
Babel                              2.8.0
backcall                           0.2.0
backports.functools-lru-cache      1.6.1
backports.shutil-get-terminal-size 1.0.0
backports.tempfile                 1.0
backports.weakref                  1.0.post1
base58                             2.0.1
bcrypt                             3.2.0
beautifulsoup4                     4.9.3
binaryornot                        0.4.4
bitarray                           1.4.0
bkcharts                           0.2
black                              20.8b1
bleach                             3.1.5
blinker                            1.4
blis                               0.7.5
bokeh                              2.1.1
boto                               2.49.0
boto3                              1.20.29
botocore                           1.23.29
Bottleneck                         1.3.2
Brotli                             1.0.9
brotlipy                           0.7.0
cachetools                         4.1.1
catalogue                          2.0.6
catboost                           0.24.2
celluloid                          0.2.0
certifi                            2020.6.20
cffi                               1.14.3
chardet                            3.0.4
charset-normalizer                 2.0.10
click                              7.1.2
click-plugins                      1.1.1
cligj                              0.5.0
cloudpickle                        1.5.0
clyent                             1.2.2
cmdstanpy                          0.9.68
colorama                           0.4.3
commonmark                         0.9.1
compound-types                     0.0.5
conda                              4.8.5
conda-package-handling             1.7.0
conda-verify                       3.4.2
configparser                       5.0.1
contextlib2                        0.6.0.post1
convertdate                        2.4.0
cookiecutter                       1.7.2
cop-mlflow                         0.13.11100
copawscli                          0.1.5
coverage                           5.5
cpyai                              0.3.2
cryptography                       2.9.2
cx-Oracle                          8.1.0
cycler                             0.10.0
cymem                              2.0.6
Cython                             0.29.21
cytoolz                            0.10.1
darts                              0.17.1
dash                               1.16.3
dash-core-components               1.12.1
dash-html-components               1.1.1
dash-renderer                      1.8.2
dash-table                         4.10.1
dask                               2.20.0
databricks-cli                     0.12.2
deap                               1.3.1
decorator                          4.4.2
defusedxml                         0.6.0
descartes                          1.1.0
diff-match-patch                   20200713
dill                               0.3.2
diskcache                          5.2.1
distributed                        2.20.0
dm-tree                            0.1.6
docker                             4.3.1
docutils                           0.16
eli5                               0.11.0
entrypoints                        0.3
enum-compat                        0.0.3
ephem                              4.1.3
esda                               2.3.6
et-xmlfile                         1.0.1
fastcache                          1.1.0
filelock                           3.0.12
Fiona                              1.8.13.post1
fire                               0.3.1
flake8                             3.8.3
Flask                              1.1.2
Flask-Compress                     1.7.0
flatbuffers                        1.12
fsspec                             2022.1.0
future                             0.18.2
gast                               0.4.0
GDAL                               3.0.4
gensim                             3.8.3
geopandas                          0.8.1
gevent                             20.6.2
giddy                              2.3.3
gitdb                              4.0.5
GitPython                          3.1.9
glob2                              0.7
gmpy2                              2.0.8
google-auth                        1.22.1
google-auth-oauthlib               0.4.1
google-pasta                       0.2.0
gorilla                            0.3.0
graphviz                           0.14.2
greenlet                           0.4.16
grpcio                             1.34.1
gunicorn                           20.0.4
h5py                               3.1.0
HeapDict                           1.0.1
hijri-converter                    2.2.3
holidays                           0.13
html5lib                           1.1
huggingface-hub                    0.4.0
hyperopt                           0.2.5
idna                               2.10
imageio                            2.9.0
imagesize                          1.2.0
importlib-metadata                 1.7.0
inequality                         1.0.0
intervaltree                       3.0.2
ipykernel                          5.3.2
ipython                            7.16.1
ipython-genutils                   0.2.0
ipywidgets                         7.5.1
isodate                            0.6.0
isort                              4.3.21
itsdangerous                       1.1.0
jdcal                              1.4.1
jedi                               0.17.1
jeepney                            0.4.3
jenkspy                            0.1.4
Jinja2                             2.11.2
jinja2-time                        0.2.0
jmespath                           0.10.0
joblib                             0.16.0
json5                              0.9.5
jsonschema                         3.2.0
jupyter                            1.0.0
jupyter-client                     6.1.6
jupyter-console                    6.1.0
jupyter-core                       4.6.3
jupyterlab                         2.1.5
jupyterlab-server                  1.2.0
Keras                              2.4.3
keras-nightly                      2.5.0.dev2021032900
Keras-Preprocessing                1.1.2
kiwisolver                         1.2.0
korean-lunar-calendar              0.2.1
langcodes                          3.3.0
lasio                              0.28
lazy-object-proxy                  1.4.3
libarchive-c                       2.9
libpysal                           4.4.0
lightgbm                           3.0.0
llvmlite                           0.31.0
locket                             0.2.0
LunarCalendar                      0.0.9
lxml                               4.6.3
lz4                                3.1.0
Mako                               1.1.3
mamba                              0.1.2
mapclassify                        2.4.2
Markdown                           3.3.1
MarkupSafe                         1.1.1
matplotlib                         3.4.1
mccabe                             0.6.1
mgwr                               2.1.2
missingno                          0.4.2
mistune                            0.8.4
mkl-fft                            1.1.0
mkl-random                         1.1.1
mkl-service                        2.3.0
mlflow                             1.11.0
mock                               4.0.2
more-itertools                     8.4.0
mpl-format                         0.297
mpmath                             1.1.0
msgpack                            1.0.0
msrest                             0.6.19
multidict                          4.7.6
multipledispatch                   0.6.0
munch                              2.5.0
murmurhash                         1.0.6
mypy-extensions                    0.4.3
navigator-updater                  0.2.1
nbclient                           0.5.0
nbconvert                          5.6.1
nbformat                           5.0.7
nest-asyncio                       1.4.1
networkx                           2.4
nfoursid                           1.0.0
nltk                               3.5
nose                               1.3.7
notebook                           6.0.3
numba                              0.48.0
numexpr                            2.7.1
numpy                              1.19.5
numpydoc                           1.1.0
oauthlib                           3.1.0
olefile                            0.46
opencv-contrib-python              4.5.2.52
openpyxl                           3.0.4
opt-einsum                         3.3.0
packaging                          20.4
pandas                             1.1.4
pandera                            0.6.3
pandocfilters                      1.4.2
papermill                          2.2.0
paramiko                           2.7.2
parso                              0.7.0
partd                              1.1.0
path                               13.1.0
pathlib2                           2.3.5
pathspec                           0.8.0
pathtools                          0.1.2
pathy                              0.6.1
patsy                              0.5.2
pep8                               1.7.1
pexpect                            4.8.0
pickleshare                        0.7.5
Pillow                             7.2.0
pip                                20.1.1
pkginfo                            1.5.0.1
plotly                             4.11.0
pluggy                             0.13.1
ply                                3.11
pmdarima                           1.8.4
pointpats                          2.2.0
poyo                               0.5.0
preshed                            3.0.6
probability                        0.0.153
prometheus-client                  0.8.0
prometheus-flask-exporter          0.18.1
prompt-toolkit                     3.0.5
prophet                            1.0.1
protobuf                           3.13.0
psutil                             5.7.0
psycopg2-binary                    2.8.6
ptyprocess                         0.6.0
PuLP                               2.4
py                                 1.9.0
py4j                               0.10.9
pyaml                              20.4.0
pyarrow                            1.0.1
pyasn1                             0.4.8
pyasn1-modules                     0.2.8
pyathena                           2.2.0
pycodestyle                        2.6.0
pycosat                            0.6.3
pycparser                          2.20
pycryptodome                       3.10.1
pycurl                             7.43.0.5
pydantic                           1.8.2
pydeck                             0.5.0b1
pyDeprecate                        0.3.1
pydocstyle                         5.0.2
pyerfa                             2.0.0.1
pyflakes                           2.2.0
pygeos                             0.10
Pygments                           2.6.1
PyJWT                              1.7.1
pylint                             2.5.3
PyMeeus                            0.5.11
Pympler                            1.0.1
PyNaCl                             1.4.0
pyodbc                             4.0.0-unsupported
pyOpenSSL                          19.1.0
pyparsing                          2.4.7
pyproj                             2.6.1.post1
pyrsistent                         0.16.0
pysal                              2.4.0
PySocks                            1.7.1
pyspark                            3.0.1
pystan                             2.19.1.1
pytest                             5.4.3
pytest-cov                         2.12.0
python-dateutil                    2.8.0
python-dotenv                      0.15.0
python-editor                      1.0.4
python-jsonrpc-server              0.3.4
python-language-server             0.34.1
python-slugify                     4.0.1
pytorch-lightning                  1.5.10
pytz                               2020.1
PyWavelets                         1.1.1
pyxdg                              0.26
PyYAML                             5.3.1
pyzmq                              19.0.1
QDarkStyle                         2.8.1
QtAwesome                          0.7.2
qtconsole                          4.7.5
QtPy                               1.9.0
quantecon                          0.5.0
querystring-parser                 1.2.4
rasterio                           1.2.3
rasterstats                        0.15.0
regex                              2020.6.8
requests                           2.27.1
requests-oauthlib                  1.3.0
retrying                           1.3.3
rich                               10.4.0
rope                               0.17.0
rpy2                               3.3.6
rsa                                4.6
Rtree                              0.9.4
ruamel-yaml                        0.15.87
s3transfer                         0.5.0
sacremoses                         0.0.47
scikit-gstat                       0.5.6
scikit-image                       0.16.2
scikit-learn                       1.0.2
scikit-optimize                    0.9.0
scipy                              1.6.3
seaborn                            0.10.1
SecretStorage                      3.1.2
segregation                        1.5.0
segyio                             1.9.3
Send2Trash                         1.5.0
setuptools                         59.5.0
setuptools-git                     1.2
shap                               0.39.0
Shapely                            1.7.1
simplegeneric                      0.8.1
simplejson                         3.17.2
singledispatch                     3.4.0.3
sip                                4.19.13
six                                1.15.0
slicer                             0.0.7
smart-open                         5.2.1
smmap                              3.0.4
snowballstemmer                    2.0.0
snuggs                             1.4.7
sortedcollections                  1.2.1
sortedcontainers                   2.2.2
soupsieve                          2.0.1
spacy                              3.2.2
spacy-legacy                       3.0.9
spacy-loggers                      1.0.1
spacy-lookups-data                 1.0.3
spaghetti                          1.5.6
spglm                              1.0.8
Sphinx                             3.1.2
sphinxcontrib-applehelp            1.0.2
sphinxcontrib-devhelp              1.0.2
sphinxcontrib-htmlhelp             1.0.3
sphinxcontrib-jsmath               1.0.1
sphinxcontrib-qthelp               1.0.3
sphinxcontrib-serializinghtml      1.1.4
sphinxcontrib-websupport           1.2.3
spint                              1.0.7
splot                              1.1.3
spopt                              0.1.1
spreg                              1.2.2
spvcm                              0.3.0
spyder-kernels                     1.9.2
SQLAlchemy                         1.3.13
sqlalchemy-redshift                0.8.2
sqlparams                          3.0.0
sqlparse                           0.4.1
srsly                              2.4.2
statsmodels                        0.13.0
stopit                             1.1.2
streamlit                          1.5.0
sympy                              1.6.1
tables                             3.6.1
tabulate                           0.8.7
tblib                              1.6.0
tenacity                           6.2.0
tensorboard                        2.5.0
tensorboard-data-server            0.6.1
tensorboard-plugin-wit             1.6.0.post3
tensorflow                         2.5.0
tensorflow-estimator               2.5.0
tensorflow-probability             0.12.2
teradata                           15.10.0.21
teradatasql                        17.0.0.8
teradatasqlalchemy                 17.0.0.1
termcolor                          1.1.0
terminado                          0.8.3
testpath                           0.4.4
text-unidecode                     1.3
textwrap3                          0.9.2
thinc                              8.0.13
threadpoolctl                      2.1.0
tobler                             0.7.0
tokenizers                         0.11.5
toml                               0.10.2
toolz                              0.10.0
torch                              1.10.2
torchmetrics                       0.7.2
tornado                            6.0.4
TPOT                               0.11.5
tqdm                               4.62.3
traitlets                          4.3.3
transformers                       4.16.2
trimesh                            3.9.14
typed-ast                          1.4.1
typer                              0.4.0
typing-extensions                  3.7.4.3
typing-inspect                     0.6.0
tzlocal                            2.1
ujson                              1.35
unicodecsv                         0.14.1
Unidecode                          1.2.0
update-checker                     0.18.0
urllib3                            1.25.11
validators                         0.18.1
vedo                               2021.0.3
vtk                                9.0.1
wasabi                             0.9.0
watchdog                           0.10.3
wcwidth                            0.2.5
webencodings                       0.5.1
websocket-client                   0.57.0
Werkzeug                           1.0.1
wheel                              0.36.2
whichcraft                         0.6.1
widgetsnbextension                 3.5.1
wrapt                              1.12.1
wurlitzer                          2.0.1
xarray                             0.21.1
xgboost                            1.2.0
xlrd                               1.2.0
XlsxWriter                         1.2.9
xlwt                               1.3.0
xmltodict                          0.12.0
yapf                               0.30.0
yarl                               1.5.1
zict                               2.0.0
zipp                               3.1.0
zope.event                         4.4
zope.interface                     4.7.1

**Comment by mauicv** [comment]
Hey @Chris-Olsen, 
I've issued the fix, thanks for your help!
